rllib/execution/train_ops.py

```diff
@@ -65,6 +65,104 @@ def train_one_step(trainer, train_batch) -> Dict:
     return info
 
 
+@ExperimentalAPI
+def multi_gpu_train_one_step(trainer, train_batch) -> Dict:
+    config = trainer.config
+    workers = trainer.workers
+    local_worker = workers.local_worker()
+    policies = local_worker.policies_to_train
+    num_sgd_iter = config.get("sgd_num_iter", 1)
+    sgd_minibatch_size = config.get("sgd_minibatch_size", 0)
+
+    # Collect actual GPU devices to use.
+    if not config.get("num_gpus", 0):
+        _fake_gpus = True
+        num_gpus = 1
+    type_ = "cpu" if _fake_gpus else "gpu"
+    devices = [
+        "/{}:{}".format(type_, 0 if _fake_gpus else i)
+        for i in range(int(math.ceil(num_gpus)))
+    ]
+
+    # Make sure total batch size is dividable by the number of devices.
+    # Batch size per tower.
+    per_device_batch_size = sgd_minibatch_size // len(devices)
+    # Total batch size.
+    batch_size = per_device_batch_size * len(devices)
```

Comment:
we are computing batch_size, but not using it anywhere below.
also the first assert seems unnecessary if we are overwriting batch_size like this here?
so in this scheme, what role does the train_batch_size parameter play? we just ignore it?

Reply:
You are right. I cleaned this part up a little. There are no more unnecessary variables now.